### ISSUE TITLE
Gem::Specification#_load sets @license instead of @licenses. 

### ISSUE THREAD

**Issue by djberube** [rubygems/rubygems#7978]
### Describe the problem as clearly as you can

Gem::Specification#_load sets @license. It should set @licenses. 

This script should work; it does not:

```
require 'rubygems'

fetcher = Gem::SpecFetcher.fetcher

specs = fetcher.detect(:latest) do |tuple|

                                  ARGV.detect { |arg| tuple.name.start_with? arg}

                                end.map do |tuple| 
                                  name_tuple, source = *tuple
                                  source.fetch_spec(name_tuple)
                                end

specs.each do |_|
  puts "#{_.name} released under #{_.license}"
end
```

### Did you try upgrading RubyGems?

Yes, the bug is still present in the latest version of RubyGems.

### Post steps to reproduce the problem

Open PR to fix.

**Comment by djberube** [comment]
This is related to PR #7975.

**Comment by deivid-rodriguez** [comment]
Closed by #7975.
